core/templates/dev/head/pages/exploration_editor/ParamMetadataObjectFactorySpec.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+* @fileoverview Unit tests for ParamMetadataObjectFactory.
+*/
+
+describe('ParameterMetadata object factory', function() {
+  var parameterMetadata = null;
+  var pmof = null;
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    pmof = $injector.get('ParamMetadataObjectFactory');
+
+    parameterMetadata = pmof.createWithSetAction('answer', 'param_changes', 1);
+
+    it('should have correct metadata for SET action', function() {
```

Comment:
Why are these tests within beforeEach function?

Reply:
I have also moved `ParamMetadataObjectFactory.js` to `core/templates/dev/head/domain/exploration`

Reply:
Hi @hoangviet1993 -- I think the tests are still in the beforeEach()? Look at the other files, the it() is separated from the beforeEach() entirely.